shell/spy_test.go

```diff
@@ -1,14 +1,33 @@
 package shell
 
 import (
-	"fmt"
+	"reflect"
 	"testing"
 )
 
-func TestExecute(t *testing.T) {
-	want := fmt.Sprintf("spy.go\nspy_test.go\n")
-	got := Execute("ls")
-	if got != want {
-		t.Errorf("Expected: %s, got: %s", want, got)
+func TestParseCommand(t *testing.T) {
+	testData := []struct {
+		cmd  string
+		name string
+		args []string
+	}{
+		{"echo Bukola", "echo", []string{"Bukola"}},
+		{"test it works fine", "test", []string{"it", "works", "fine"}},
+		{"cat /file-path", "cat", []string{"/file-path"}},
+		{"cd ..", "cd", []string{".."}},
+		{"echo", "echo", []string{}},
 	}
+
+	for _, td := range testData {
+
+		name, args := ParseCommand(td.cmd)
+
+		if td.name != name {
+			t.Errorf("Expected command name: %s, got: %s", td.name, name)
+		}
+		if !reflect.DeepEqual(td.args, args) {
```

Comment:
We can use `cmp.Equal` here (from the [go-cmp](https://github.com/google/go-cmp) library, which is not only a bit easier to use than `reflect`, it has a nice `cmp.Diff` function to diff the two slices.